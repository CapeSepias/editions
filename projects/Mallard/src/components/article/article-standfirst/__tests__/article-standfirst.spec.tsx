import React from 'react'
import TestRenderer, { ReactTestRendererJSON } from 'react-test-renderer'
import { ArticleStandfirst } from '..'

describe('ArticleStandfirst', () => {
    it('should display a default Standfirst', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <ArticleStandfirst standfirst="This is a test Standfirst" />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
    it('should be able to asborb an injected style', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <ArticleStandfirst standfirst="This is a test Standfirst" />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
})
