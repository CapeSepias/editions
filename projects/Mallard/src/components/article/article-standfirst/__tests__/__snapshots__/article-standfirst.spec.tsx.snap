// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleStandfirst should be able to asborb an injected style 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "flexShrink": 0,
          "fontFamily": "GuardianTextEgyptian-Reg",
          "fontSize": 17,
          "lineHeight": 21,
        },
        Array [
          Object {
            "color": "#121212",
          },
          Object {},
        ],
      ]
    }
  >
    This is a test Standfirst
  </Text>
  <View
    style={
      Array [
        Object {
          "marginBottom": 12,
          "marginTop": 12,
          "paddingTop": 3,
          "width": "100%",
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#dcdcdc",
        },
        Object {
          "alignContent": "stretch",
          "alignItems": "stretch",
          "justifyContent": "flex-end",
        },
      ]
    }
  />
</View>
`;

exports[`ArticleStandfirst should display a default Standfirst 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "flexShrink": 0,
          "fontFamily": "GuardianTextEgyptian-Reg",
          "fontSize": 17,
          "lineHeight": 21,
        },
        Array [
          Object {
            "color": "#121212",
          },
          Object {},
        ],
      ]
    }
  >
    This is a test Standfirst
  </Text>
  <View
    style={
      Array [
        Object {
          "marginBottom": 12,
          "marginTop": 12,
          "paddingTop": 3,
          "width": "100%",
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#dcdcdc",
        },
        Object {
          "alignContent": "stretch",
          "alignItems": "stretch",
          "justifyContent": "flex-end",
        },
      ]
    }
  />
</View>
`;
